pkg/provider/azure_vmssflex_cache.go

```diff
@@ -124,27 +125,73 @@ func (fs *FlexScaleSet) newVmssFlexVMCache() (*azcache.TimedCache, error) {
 	return azcache.NewTimedcache(time.Duration(fs.Config.VmssFlexVMCacheTTLInSeconds)*time.Second, getter)
 }
 
+func (fs *FlexScaleSet) getVmssFlexIDByComputerNamePrefix(computerNamePrefix string, crt azcache.AzureCacheReadType) (string, error) {
+	targetVmssFlexID := ""
+	cached, err := fs.vmssFlexCache.Get(consts.VmssFlexKey, crt)
+	if err != nil {
+		return "", err
+	}
+	vmssFlexes := cached.(*sync.Map)
+	vmssFlexes.Range(func(key, value interface{}) bool {
+		vmssFlex := value.(*compute.VirtualMachineScaleSet)
+
+		if vmssFlex.VirtualMachineProfile != nil &&
+			vmssFlex.VirtualMachineProfile.OsProfile != nil &&
+			vmssFlex.VirtualMachineProfile.OsProfile.ComputerNamePrefix != nil &&
+			strings.EqualFold(*vmssFlex.VirtualMachineProfile.OsProfile.ComputerNamePrefix, computerNamePrefix) {
+			targetVmssFlexID = key.(string)
+			return false
+		}
+		return true
+	})
+	return targetVmssFlexID, nil
+}
+
 func (fs *FlexScaleSet) getNodeVmssFlexID(nodeName string) (string, error) {
 	fs.lockMap.LockEntry(consts.GetNodeVmssFlexIDLockKey)
 	defer fs.lockMap.UnlockEntry(consts.GetNodeVmssFlexIDLockKey)
-	vmssFlexID, isCached := fs.vmssFlexVMNameToVmssID.Load(nodeName)
-	if !isCached {
-		klog.V(12).Infof("nodeName %s is not saved in vmssFlexVMnameToVmssID map, send a GET request to retrieve its VmssID", nodeName)
-		machine, err := fs.getVirtualMachine(types.NodeName(nodeName), azcache.CacheReadTypeUnsafe)
+	cachedVmssFlexID, isCached := fs.vmssFlexVMNameToVmssID.Load(nodeName)
+
+	if isCached {
+		return fmt.Sprintf("%v", cachedVmssFlexID), nil
+	}
+
+	vmssFlexID := ""
+	var err error
+	if fs.Config.VmssFlexUsingDefaultVMProfile != nil && *fs.Config.VmssFlexUsingDefaultVMProfile {
```

Comment:
we can use to.Bool here

Reply:
Thanks for the suggestion. Just changed it.